packages/connection/src/Client.php

```diff
@@ -0,0 +1,400 @@
+<?php
+
+namespace Automattic\Jetpack\Connection;
+
+class Client {
+	const WPCOM_JSON_API_VERSION = '1.1';
+
+	/**
+	 * Makes an authorized remote request using Jetpack_Signature
+	 *
+	 * @return array|WP_Error WP HTTP response on success
+	 */
+	public static function remote_request( $args, $body = null ) {
+		$defaults = array(
+			'url'           => '',
+			'user_id'       => 0,
+			'blog_id'       => 0,
+			'auth_location' => JETPACK_CLIENT__AUTH_LOCATION,
+			'method'        => 'POST',
+			'timeout'       => 10,
+			'redirection'   => 0,
+			'headers'       => array(),
+			'stream'        => false,
+			'filename'      => null,
+			'sslverify'     => true,
+		);
+
+		$args = wp_parse_args( $args, $defaults );
+
+		$args['blog_id'] = (int) $args['blog_id'];
+
+		if ( 'header' != $args['auth_location'] ) {
+			$args['auth_location'] = 'query_string';
+		}
+
+		$token = \Jetpack_Data::get_access_token( $args['user_id'] );
+		if ( ! $token ) {
+			return new Jetpack_Error( 'missing_token' );
+		}
+
+		$method = strtoupper( $args['method'] );
+
+		$timeout = intval( $args['timeout'] );
+
+		$redirection = $args['redirection'];
+		$stream      = $args['stream'];
+		$filename    = $args['filename'];
+		$sslverify   = $args['sslverify'];
+
+		$request = compact( 'method', 'body', 'timeout', 'redirection', 'stream', 'filename', 'sslverify' );
+
+		@list( $token_key, $secret ) = explode( '.', $token->secret );
+		if ( empty( $token ) || empty( $secret ) ) {
+			return new Jetpack_Error( 'malformed_token' );
+		}
+
+		$token_key = sprintf( '%s:%d:%d', $token_key, JETPACK__API_VERSION, $token->external_user_id );
+
+		$time_diff         = (int) \Jetpack_Options::get_option( 'time_diff' );
+		$jetpack_signature = new \Jetpack_Signature( $token->secret, $time_diff );
+
+		$timestamp = time() + $time_diff;
+
+		if ( function_exists( 'wp_generate_password' ) ) {
+			$nonce = wp_generate_password( 10, false );
+		} else {
+			$nonce = substr( sha1( rand( 0, 1000000 ) ), 0, 10 );
+		}
+
+		// Kind of annoying.  Maybe refactor Jetpack_Signature to handle body-hashing
+		if ( is_null( $body ) ) {
+			$body_hash = '';
+
+		} else {
+			// Allow arrays to be used in passing data.
+			$body_to_hash = $body;
+
+			if ( is_array( $body ) ) {
+				// We cast this to a new variable, because the array form of $body needs to be
+				// maintained so it can be passed into the request later on in the code.
+				if ( count( $body ) > 0 ) {
+					$body_to_hash = json_encode( self::_stringify_data( $body ) );
+				} else {
+					$body_to_hash = '';
+				}
+			}
+
+			if ( ! is_string( $body_to_hash ) ) {
+				return new \Jetpack_Error( 'invalid_body', 'Body is malformed.' );
+			}
+
+			$body_hash = Jetpack::connection()->sha1_base64( $body_to_hash );
```

Comment:
Now that this is in the connection package, should it reference the connection in a different way? After all, this couples the functionality to the main Jetpack class, which we definitely don't want to do.

Or at minimum, to make this work temporarily, we need to prefix `Jetpack` with `\`.

Reply:
I think just replacing the entire call with the method body is going to be best here.